syweb/webclient/test/unit/notification-service.spec.js

```diff
@@ -1,78 +1,340 @@
-        expect(notificationService.containsBingWord(userId, displayName, 
-        bingWords, "Hello @ali, how are you?")).toEqual(true);
+        testEvent.content.body = "Hello ali, how are you?";
+        expect(notificationService.shouldHighlightEvent(testEvent)).toEqual(true);
     }));
     
     it('should bing on a case-insensitive user ID.', inject(
     function(notificationService) {
-        expect(notificationService.containsBingWord(userId, displayName, 
-        bingWords, "Hello @AlI:matrix.org, how are you?")).toEqual(true);
+        testEvent.content.body = "Hello @AlI:matrix.org, how are you?";
+        expect(notificationService.shouldHighlightEvent(testEvent)).toEqual(true);
     }));
     
     // Display names
     
     it('should bing on a display name.', inject(
     function(notificationService) {
-        expect(notificationService.containsBingWord(userId, displayName, 
-        bingWords, "Hello Alice M, how are you?")).toEqual(true);
+        testEvent.content.body = "Hello Alice M, how are you?";
+        expect(notificationService.shouldHighlightEvent(testEvent)).toEqual(true);
     }));
     
     it('should bing on a case-insensitive display name.', inject(
     function(notificationService) {
-        expect(notificationService.containsBingWord(userId, displayName, 
-        bingWords, "Hello ALICE M, how are you?")).toEqual(true);
+        testEvent.content.body = "Hello ALICE M, how are you?";
+        expect(notificationService.shouldHighlightEvent(testEvent)).toEqual(true);
     }));
     
     // Bing words
     
     it('should bing on a bing word.', inject(
     function(notificationService) {
-        expect(notificationService.containsBingWord(userId, displayName, 
-        bingWords, "I really like coffee")).toEqual(true);
+        testEvent.content.body = "I really like coffee";
+        expect(notificationService.shouldHighlightEvent(testEvent)).toEqual(true);
     }));
     
     it('should bing on case-insensitive bing words.', inject(
     function(notificationService) {
-        expect(notificationService.containsBingWord(userId, displayName, 
-        bingWords, "Coffee is great")).toEqual(true);
+        testEvent.content.body = "Coffee is great";
+        expect(notificationService.shouldHighlightEvent(testEvent)).toEqual(true);
     }));
     
     it('should bing on wildcard (.*) bing words.', inject(
     function(notificationService) {
-        expect(notificationService.containsBingWord(userId, displayName, 
-        bingWords, "It was foomahbar I think.")).toEqual(true);
+        testEvent.content.body = "It was foomahbar I think.";
+        expect(notificationService.shouldHighlightEvent(testEvent)).toEqual(true);
+    }));
+
+    it('should bing on character group ([abc]) bing words.', inject(
+    function(notificationService) {
+        testEvent.content.body = "Ping!";
+        expect(notificationService.shouldHighlightEvent(testEvent)).toEqual(true);
```

Comment:
There's an unfortunate disconnect between the configuration of the push rules (from the mocked ``matrixService``) and the tests here. It would be much better if you could configure the push rules in the test then test that ``notificationService`` was doing the right thing. As it stands, these tests seem very magicy without the context of the push rules (which are now really large and difficult to map to an individual test). This will require lazily loading the push rules if you aren't already, so the act of instantiating the ``notificationService`` doesn't do the ``GET``.